baseplate/experiments/__init__.py

```diff
@@ -77,6 +78,16 @@ def _get_config(self, name):
             logger.warning("Could not load experiment config: %s", str(exc))
             return
 
+    def _get_experiment(self, name):
+        if name not in self._experiment_cache:
```

Comment:
No, nothing invalidates it so it does cache away the use of the filewatcher.  My thinking was you get a new `Experiments` object for each request so you're only caching away the file watcher for a single request.